### PR TITLE
Isolate parameter handling to ToolDataSink class

### PR DESCRIPTION
**Pull request by portante**

### CODE REVIEW

**Review by dbutenhof — APPROVED** [review]

File: lib/pbench/agent/tool_data_sink.py [hunk 1]
```diff
@@ -1143,22 +1160,22 @@ def record_tms(self, tms):
 
         section = "pbench"
         mdlog.add_section(section)
-        mdlog.set(section, "config", self.optional_md["config"])
-        mdlog.set(section, "date", self.optional_md["date"])
-        mdlog.set(section, "name", self.params.benchmark_run_dir.local.name)
+        mdlog.set(section, "config", self.optional_md.get("config", ""))
```
Comment (portante) [inline lib/pbench/agent/tool_data_sink.py:1163]: This represents a real bug.  I'll file a direct change to `main`.

Comment (portante) [reply lib/pbench/agent/tool_data_sink.py:1163]: Filed PR #2770.

**Review by portante — COMMENTED** [review]
Can we consider this?

File: lib/pbench/agent/tool_data_sink.py (hunk 1 above)
Comment (webbnh) [reply lib/pbench/agent/tool_data_sink.py:1163]: Can `self.optional_md` itself be `None`, or is it always at least an empty dictionary?  (I'll try to figure this out, but I figured I would ask first.)

Comment (webbnh) [reply lib/pbench/agent/tool_data_sink.py:1163]: OK, nevermind:  it's set to an empty dictionary if it's otherwise unspecified.

**Pull request closed without merging**